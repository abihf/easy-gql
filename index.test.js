const gql = require('graphql-tag');
const egql = require('./index');

describe('easy-gql', () => {
  it('should return same AST as vanilla graphql-tag', () => {
    const gqlQuery = gql`query Q1 { test }`;
    const egqlQuery = egql`query Q1 { test }`;
    expect(egqlQuery).toMatchObject(gqlQuery);
  });

  it('should handle simple fragment', () => {
    const gqlFragment = gql`fragment F1 on Type { field2 }`
    const gqlQuery = gql`query Q1 { test { ...F1 } } ${gqlFragment}`;
    delete gqlQuery.loc;

    const egqlFragment = egql`fragment F1 on Type { field2 }`
    const egqlQuery = egql`query Q1 { test { ...${gqlFragment} } }`;

    expect(egqlQuery).toMatchObject(gqlQuery);
  });

  it('should handle multiple fragment', () => {
    const fragment1 = egql`fragment F2 on ParentType { field1 }`
    const childFragment = egql`fragment FC on ChildType { childFragment }`
    const fragment2 = egql`fragment F3 on ParentType { field2 ...${childFragment} }`
    const query = egql`query Q1 { parent { ...${fragment1} ...${fragment2} }}`;

    expect(query).toMatchSnapshot();
  })

})
