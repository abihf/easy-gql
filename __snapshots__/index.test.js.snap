// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`easy-gql should handle multiple fragment 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "Q1",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "parent",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "directives": Array [],
                  "kind": "FragmentSpread",
                  "name": Object {
                    "kind": "Name",
                    "value": "F2",
                  },
                },
                Object {
                  "directives": Array [],
                  "kind": "FragmentSpread",
                  "name": Object {
                    "kind": "Name",
                    "value": "F3",
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
    Object {
      "directives": Array [],
      "kind": "FragmentDefinition",
      "name": Object {
        "kind": "Name",
        "value": "F2",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "field1",
            },
            "selectionSet": null,
          },
        ],
      },
      "typeCondition": Object {
        "kind": "NamedType",
        "name": Object {
          "kind": "Name",
          "value": "ParentType",
        },
      },
    },
    Object {
      "directives": Array [],
      "kind": "FragmentDefinition",
      "name": Object {
        "kind": "Name",
        "value": "F3",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "field2",
            },
            "selectionSet": null,
          },
          Object {
            "directives": Array [],
            "kind": "FragmentSpread",
            "name": Object {
              "kind": "Name",
              "value": "FC",
            },
          },
        ],
      },
      "typeCondition": Object {
        "kind": "NamedType",
        "name": Object {
          "kind": "Name",
          "value": "ParentType",
        },
      },
    },
    Object {
      "directives": Array [],
      "kind": "FragmentDefinition",
      "name": Object {
        "kind": "Name",
        "value": "FC",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "childFragment",
            },
            "selectionSet": null,
          },
        ],
      },
      "typeCondition": Object {
        "kind": "NamedType",
        "name": Object {
          "kind": "Name",
          "value": "ChildType",
        },
      },
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 34,
    "start": 0,
  },
}
`;
